File: implementation/src/benchmark/kotlin/com/aoc/intcode/tractorbeam/DroneSystemTest.java
package com.aoc.intcode.tractorbeam;

import com.aoc.AdventBenchmarkingSuite;
import com.aoc.BenchmarkInputReader;
import com.aoc.Day;
import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Warmup;

@AdventBenchmarkingSuite
public class DroneSystemTest {
    @Benchmark
    @Warmup(iterations = 1)
    @Fork(value = 1, warmups = 1)
    public void solutionPartTwo() {
        final String input = BenchmarkInputReader.read(new Day(19)).asSingleString();
        new DroneSystem(input).scanAreaForSantasShip(100);
    }
}

File: implementation/src/benchmark/kotlin/com/aoc/intcode/tractorbeam/DroneSystemTest.kt
package com.aoc.intcode.tractorbeam

import com.aoc.input.Day
import com.aoc.input.InputReader
import org.junit.jupiter.api.Test
import org.openjdk.jmh.annotations.Benchmark
import kotlin.math.abs

class DroneSystemTest {
    @Test
    @Benchmark
    fun solutionPartTwo() {
        val input = InputReader().readInputAsSingleString(Day.from(19))
        val system = DroneSystem(input).scanAreaForSantasShip(100)
        abs(1)
    }
}
